gdal/frmts/derived/deriveddataset.cpp

```diff
@@ -192,6 +192,12 @@ GDALDataset * DerivedDataset::Open(GDALOpenInfo * poOpenInfo)
 
     GDALClose(poTmpDS);
 
+    CPLString path = CPLGetPath(odFilename);
+    CPLString ovrFileName = "DERIVED_DATASET_"+odDerivedName+"_"+CPLGetFilename(odFilename);
+    CPLString ovrFilePath = CPLFormFilename(path,ovrFileName,NULL);
```

Comment:
This only makes sense if odFilename points to a "real" file, but GDAL dataset names can be more diverse than that. So I think a VSIStatL(odFilename, &sStatBuf) == 0 test should be done before trying to initialize the overview manager.